sqlmesh/core/engine_adapter/databricks.py

```diff
@@ -47,12 +46,20 @@ def __init__(self, *args: t.Any, **kwargs: t.Any):
         super().__init__(*args, **kwargs)
         self._spark: t.Optional[PySparkSession] = None
 
-    @classproperty
-    def can_access_spark_session(cls) -> bool:
+    @classmethod
+    def can_access_spark_session(cls, disable_spark_session: bool) -> bool:
         from sqlmesh import RuntimeEnv
 
-        if RuntimeEnv.get().is_databricks:
-            return True
+        if disable_spark_session:
```

Comment:
This seems counterintuitive to me. IMHO if `disable_spark_session` is False we shouldn't even call this method, no? What's the point of calling it to immediately return from it?

Reply:
To make sure the caller considers the possibility of the spark session being disabled. It is easy to just say if I'm in a notebook then use SparkSession and so I wanted to make sure when someone checks for this we bundle in the check for disabling spark session.

Reply:
I can't rely on state of the class since we do this check from outside and inside the class (back in connection config).